specification/storage/resource-manager/Microsoft.Storage/stable/2021-04-01/file.json

```diff
@@ -0,0 +1,1094 @@
+          },
+          {
+            "name": "$maxpagesize",
+            "in": "query",
+            "type": "string",
+            "description": "Optional. Specified maximum number of shares that can be included in the list."
+          },
+          {
+            "name": "$filter",
+            "in": "query",
+            "type": "string",
+            "description": "Optional. When specified, only share names starting with the filter will be listed."
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
+            "description": "Optional, used to expand the properties within share's properties. Valid values are: deleted, snapshots. Should be passed as a string with delimiter ','"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK -- List Shares operation completed successfully.",
+            "schema": {
+              "$ref": "#/definitions/FileShareItems"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Storage/storageAccounts/{accountName}/fileServices/default/shares/{shareName}": {
+      "put": {
+        "tags": [
+          "FileShares"
+        ],
+        "operationId": "FileShares_Create",
+        "x-ms-examples": {
+          "PutShares": {
+            "$ref": "./examples/FileSharesPut.json"
+          },
+          "Create NFS Shares": {
+            "$ref": "./examples/FileSharesPut_NFS.json"
+          },
+          "PutShares with Access Tier": {
+            "$ref": "./examples/FileSharesPut_AccessTier.json"
+          }
+        },
+        "description": "Creates a new share under the specified account as described by request body. The share resource includes metadata and properties for that share. It does not include a list of the files contained by the share. ",
+        "parameters": [
+          {
+            "$ref": "./storage.json#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "./storage.json#/parameters/StorageAccountName"
+          },
+          {
+            "$ref": "#/parameters/ShareName"
+          },
+          {
+            "name": "fileShare",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/FileShare"
+            },
+            "description": "Properties of the file share to create."
+          },
+          {
+            "name": "$expand",
+            "in": "query",
+            "required": false,
+            "type": "string",
```

Comment:
>"string", [](http://example.com/codeflow?start=20&length=9)

Same question, why make this an unrestricted string rather than string enum?

Reply:
Because we would want to support multiple enum strings being passed in the expand parameters. Modeling it as enum will restrict the client SDK to do so.